packages/xstate-react/src/useMachine.ts

```diff
@@ -113,6 +114,10 @@ export function useMachine<
   Interpreter<TContext, any, TEvent, TTypestate>['send'],
   Interpreter<TContext, any, TEvent, TTypestate>
 ] {
+  const machine = useMemo(() => {
+    return typeof getMachine === 'function' ? getMachine() : getMachine;
```

Comment:
Remember that `useMemo` doesn't have any semantic guarantees about the lifetime of its cache. This should be resolved within `useConstant`'s callback.